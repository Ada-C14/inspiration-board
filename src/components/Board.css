.board-container {
  margin-left: 5%;
}

.board-container__nav {
  margin-top: 2%;
  top: 0;
  position: fixed;
  display: flex;
  font-size: smaller;
  align-items: center;
  z-index: 1;
}

.board-container__nav span {
  white-space: nowrap;
  margin-right: 15px;
}

.board-container__left-side {
  position: fixed;
  width: 25%;
  margin-right: 10% !important;
}

.board-container__content {
  margin-top: 5%;
}

.board-container__content__cards {
  margin-left: 30%;
  width: 70%;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}

.alert-display {
  font-weight: 500;
  margin-bottom: 1.5em;
}

.alert-display span {
  background-color: #ffc94b;
}

@media (min-width: 320px) and (max-width: 480px) {
  .board-container {
    margin-left: 10%;
  }
  
  .board-container__nav {
    position: static;
    display: inline;
  }
}