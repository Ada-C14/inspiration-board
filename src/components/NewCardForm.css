/* The Modal (background) */

/* * {
  border: black 1px solid;
} */

.new-card-form__container {
  display: flex;
  justify-content: center;
  align-items: center;
}

.new-card-form {
  background-color: #fcf6bd;

  padding: 1em 0;
  margin: 0.5rem;

  height: 250px;
  width: 225px;
  flex-basis: 18%;

  border-radius: 5px;

  align-items: center;
}

.new-card-form__header {
  text-align: center;
}

.new-card-form__form {
  font-size: 1.5em;
  display: grid;
  grid-template-columns: [labels] auto [controls] 1fr;
  grid-auto-flow: row;
  grid-gap: .8em;
}

.new-card-form__form-label  {
  grid-column: labels;
  grid-row: auto;
}

.new-card-form__form-select,
.new-card-form__form-textarea,
.new-card-form__form-button {
  font-family: 'Raleway', sans-serif;
  font-size: 1.25em;
  grid-column: controls;
  grid-row: auto;
}

.new-card-form__form-button {
  background-color: inherit;
  border: 1px solid black;
  font-size: 1em;
}
