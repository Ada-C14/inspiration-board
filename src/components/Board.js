import React from 'react';
import PropTypes from 'prop-types';

import './Board.css';
import Card from './Card';


const Board = (props) => {
  
  const getCards = () => {
    const processed = props.cards.map(card => {
      return(
      <Card key={card.card.id} id={card.card.id} text={card.card.text} emoji={card.card.emoji} onDelete={props.deleteCard} />
      );
    })
    return processed;
  }


  return (
    <div>
      {getCards()}
    </div>
  )
};

Board.propTypes = {

};

export default Board;
