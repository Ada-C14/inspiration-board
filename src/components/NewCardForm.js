import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';
import './NewCardForm.css';

const EMOJI_LIST = ["", "heart_eyes", "beer", "clap", "sparkling_heart", "heart_eyes_cat", "dog"]

const NewCardForm =(props) => {

  const emojiDropdown = () => {
    const emojiList = EMOJI_LIST.map((emoji, index)=>{
      return  (<option key={index} value={emoji}>{emoji}</option>);
    });
    return emojiList;
  };


  return(

    <form className="NewCardForm" onSubmit={props.onSubmitCard}>
    <label>Card Text:</label>
     <input name = "text" type="text" onChange={props.onChangeCard}/>
    <label>Emoji Input:</label>
     <select  name="emoji" onChange={props.onChangeCard}>
      {emojiDropdown()}
    </select>
    <input
        type="submit"
        value="Submit Card"
      />
    </form>
  );
};

export default NewCardForm;