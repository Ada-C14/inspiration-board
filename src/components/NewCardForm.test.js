import React from 'react';
import axios from 'axios';
import '@testing-library/jest-dom/extend-expect';
import { render, screen, act, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event';
import Board from './Board';
import NewCardForm from './NewCardForm';

jest.mock('axios');

describe('Wave 3: NewCardForm', () => {
  test('clicking on the button with text "Add Card" the form will call the callback function', () => {
    // Arrange
    const callBackFunction = jest.fn();
    render(<NewCardForm addCardCallback={callBackFunction} />);

    // Act
    const submitBtn = screen.getByText(/Add Card/i);
    userEvent.click(submitBtn);

    // Assert
    expect(callBackFunction).toHaveBeenCalled();
  });

  test('can add a card to Board thru API', async () => {
    // Arrange
    const board1 = [
      {
        "card": {
          "id": 1,
          "text": "OH yeah!!!",
          "emoji": "heart_eyes_cat"
        }
      }
    ];
    const url = 'board1'  

    // Act
    // Load the cards on a board
    axios.get.mockImplementationOnce(() =>
      Promise.resolve({ data: board1 })
    );
    const { getByTestId } = render(<Board
                                      url="https://inspiration-board.herokuapp.com/"
                                      board="boards" 
                                      card="cards"  
                                      boardName={url} />);
    await waitFor(() => getByTestId(`${url}`));

    // add a card    
    const addItem = {
      "card": {
        "id": 2,
        "text": "Work!!",
        "emoji": "dog"
      }
    }      
    await axios.post.mockImplementationOnce(() => 
      Promise.resolve({ data: addItem })
    );
    userEvent.click(screen.getByText('Add Card'))
    const inspectBoard = await waitFor(() => getByTestId(`${url}`));
    
    // Assert
    expect(inspectBoard).toHaveTextContent("Work!!");
    await act(() => Promise.resolve({ data: board1 }))
  });
})
