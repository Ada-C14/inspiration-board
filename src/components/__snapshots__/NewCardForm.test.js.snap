// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewCardForm that it matches the existing snapshot of NewCardForm 1`] = `
<DocumentFragment>
  <form
    class="NewCardForm"
  >
    <label>
      Card Text:
    </label>
    <input
      name="text"
      type="text"
    />
    <label>
      Emoji Input:
    </label>
    <select
      name="emoji"
    >
      <option
        value=""
      />
      <option
        value="heart_eyes"
      >
        heart_eyes
      </option>
      <option
        value="beer"
      >
        beer
      </option>
      <option
        value="clap"
      >
        clap
      </option>
      <option
        value="sparkling_heart"
      >
        sparkling_heart
      </option>
      <option
        value="heart_eyes_cat"
      >
        heart_eyes_cat
      </option>
      <option
        value="dog"
      >
        dog
      </option>
    </select>
    <input
      type="submit"
      value="Submit Card"
    />
  </form>
</DocumentFragment>
`;
